src/tools/rustfmt/tests/target/cfg_attribute_in_where.rs

```diff
@@ -0,0 +1,95 @@
+#![crate_type = "lib"]
+#![feature(cfg_attribute_in_where)]
+use std::marker::PhantomData;
+
+#[cfg(a)]
+trait TraitA {}
+
+#[cfg(b)]
+trait TraitB {}
+
+trait A<T>
+where
+    #[cfg = a] T: TraitA,
+    #[cfg = b] T: TraitB,
+{
+    type B<U>
+    where
+        #[cfg = a] U: TraitA,
+        #[cfg = b] U: TraitB;
+
+    fn foo<U>(&self)
+    where
+        #[cfg = a] U: TraitA,
+        #[cfg = b] U: TraitB;
+}
+
+impl<T> A<T> for T
+where
+    #[cfg = a] T: TraitA,
+    #[cfg = b] T: TraitB,
+{
+    type B<U>
+        = ()
+    where
+        #[cfg = a] U: TraitA,
+        #[cfg = b] U: TraitB;
+
+    fn foo<U>(&self)
+    where
+        #[cfg = a] U: TraitA,
+        #[cfg = b] U: TraitB,
+    {
+    }
+}
+
+struct C<T>
+where
+    #[cfg = a] T: TraitA,
+    #[cfg = b] T: TraitB,
+{
+    _t: PhantomData<T>,
+}
```

Comment:
Thanks for adding these additional test cases. Given that this is an experimental feature the formatting for this isn't too important right now, but I'm wondering if the default formatting should be more like this:

```rust
struct C<T>
where
    #[cfg = a]
    T: TraitA,
    #[cfg = b]
    T: TraitB,
{
    _t: PhantomData<T>,
}
```

and then working in support for [`inline_attribute_width`](https://github.com/rust-lang/rustfmt/blob/master/Configurations.md#inline_attribute_width). To get the formatting you've got here.

Also, would you mind adding the following test case:

```rust
struct C<T>
where
    #[cfg = a] // line comment after the attribute
    T: TraitA,
    #[cfg = b] /* block comment after the attribute */
    T: TraitB,
{
    _t: PhantomData<T>,
}
```